woocommerce/woocommerce-gutenberg-products-block
# Remove `useStoreSnackbarNotices` and interact directly with data store instead / closed (see #6411)

### Important

This PR is closed because of some complicated merge conflicts. Please check this one instead: #6411.


This PR will remove the `StoreSnackbarNoticesContext` (The value - `isSuppressed` - isn't used for the Snackbars, so we can safely remove the whole context). The notice system is provided by `@wordpress/notices` and a data store from `@wordpress/data` so we can interact directly with this store instead of confusing hooks and contexts.

This PR depends on #6159. So, that one should be merged first.

- Remove `useStoreSnackbarNotices` hook.
- Remove `StoreSnackbarNoticesContext`.
- Make `StoreSnackbarNoticesContainer` a named export, so it can be imported from `@woocommerce/base-context`.
- Update all uses of `useStoreSnackbarNotices` so that it gets data from the `core/notices` store directly.
- Use the `StoreSnackbarNoticeContainer` in the cart & checkout blocks directly without a context Provider.


Fixes #5710

### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:

1. Add a coupon to your store.
2. Add items to your cart.
3. Go to the Cart block, and apply the coupon. Ensure you see the snackbar notice for a successful coupon application.
4. **In a new tab**, open the Cart block again and remove the coupon. Ensure you see the snackbar notice for successful coupon removal.
5. Go back to the original tab and try removing the coupon, ensure you see an error message above the Cart.
6. Repeat the same steps but for the Checkout block.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
